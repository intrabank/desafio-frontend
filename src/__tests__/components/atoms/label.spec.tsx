import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Label from '../../../components/atoms/label';

describe('Label', () => {
	it('should render label component', () => {
		const tree = renderer.create(<Label htmlFor='teste'>Test</Label>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render label  with out asterisc if is not type error', () => {
		render(
			<Label state='idle' htmlFor='teste'>
				Test
			</Label>
		);

		expect(screen.getByText('Test')).toBeInTheDocument();
		expect(screen.queryByText('*')).not.toBeInTheDocument();
	});

	it('should render label  with  asterisc if type error', () => {
		render(
			<Label state='error' htmlFor='teste'>
				Test
			</Label>
		);

		expect(screen.getByText('Test')).toBeInTheDocument();
		expect(screen.getByText('*')).toBeInTheDocument();
	});
});
