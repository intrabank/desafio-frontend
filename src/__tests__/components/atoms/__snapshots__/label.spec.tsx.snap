// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render label component 1`] = `
<label
  className="c-bUsXAP c-bUsXAP-fXZMCZ-state-idle"
  htmlFor="teste"
  onMouseDown={[Function]}
>
  Test
</label>
`;
